### PR TITLE
Create participants administration

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1204

**Summary of the changes made**:
1. Added BE for participant
2.  Added FE for participant.

### CODE REVIEW

File: Servers/Administration/OJS.Servers.Administration/Controllers/Api/BaseAdminApiController.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@
 [Authorize(Roles = GlobalConstants.Roles.AdministratorOrLecturer)]
 [TypeFilter(typeof(EntityPermissionsFilter))]
 public abstract class BaseAdminApiController<TEntity, TId, TGridModel, TUpdateModel> : BaseApiController
-    where TEntity : class, IEntity<TId>
```
Comment (gogo4ds) [inline Servers/Administration/OJS.Servers.Administration/Controllers/Api/BaseAdminApiController.cs:25]: Why did you remove this constraint?

File: Services/Administration/OJS.Services.Administration.Business/IAdministrationOperationService.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 using System.Threading.Tasks;
 
 public interface IAdministrationOperationService<TEntity, in TId, TUpdateModel> : IService
-    where TEntity : class, IEntity<TId>
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/IAdministrationOperationService.cs:8]: And here

File: Services/Administration/OJS.Services.Administration.Business/Participants/Validators/ParticipantAdministrationModelValidator.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            .WithMessage("Participant id cannot be less than 0");
+
+        this.RuleFor(model => model.ContestId)
+            .GreaterThan(0)
+            .WithMessage("Contest id cannot be less or equal to 0");
+
+        this.RuleFor(model => model.ContestName)
+            .NotNull()
+            .NotEmpty()
+            .WithMessage("Contest is required.");
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/Participants/Validators/ParticipantAdministrationModelValidator.cs:22]: What if you just use `.NotEmpty`, even without a custom message?

File: Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@
 using System.Threading.Tasks;
 
 public interface IGridDataService<TEntity> : IService
-    where TEntity : class, IEntity
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs:13]: Constraint

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/IAdministrationOperationService.cs (hunk 2 above)
Comment (MTsatsarov) [reply Services/Administration/OJS.Services.Administration.Business/IAdministrationOperationService.cs:8]: Because UserProfile does not comes from IEntity<Tid>, therefore cannot be used in our workflow.

File: Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs (hunk 4 above)
Comment (MTsatsarov) [reply Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs:13]: Because UserProfile does not comes from IEntity<Tid>, therefore cannot be used in our workflow.

File: Services/Administration/OJS.Services.Administration.Business/Participants/Validators/ParticipantAdministrationModelValidator.cs (hunk 3 above)
Comment (MTsatsarov) [reply Services/Administration/OJS.Services.Administration.Business/Participants/Validators/ParticipantAdministrationModelValidator.cs]: Fixed.

File: Servers/Administration/OJS.Servers.Administration/Controllers/Api/BaseAdminApiController.cs (hunk 1 above)
Comment (MTsatsarov) [reply Servers/Administration/OJS.Servers.Administration/Controllers/Api/BaseAdminApiController.cs]: Fixed

File: Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs (hunk 4 above)
Comment (MTsatsarov) [reply Services/Administration/OJS.Services.Administration.Data/IGridDataService.cs]: Fixed

**Review by gogo4ds — APPROVED** [review]

**Review by gogo4ds — APPROVED** [review]
